src/number_theory/bernoulli.lean

```diff
@@ -153,9 +153,10 @@ begin
 end
 
 open power_series
+variables (A : Type*) [integral_domain A] [algebra ℚ A]
```

Comment:
```suggestion
variables (A : Type*) [ring A] [no_zero_divisors A] [nontrivial A] [algebra ℚ A]
```

Reply:
But I suspect you don't actually need `nontrivial`, unless I'm missing something

Reply:
Seems like this is not working because `eval_neg_hom` is only defined for `comm_ring`

Reply:
Fair enough, I guess the `ring` should still be `comm_ring`.

Can you try removing the `nontrivial` assumption?

Reply:
even with `[comm_ring A] [no_zero_divisors A] [nontrivial A] ` this does not work, see https://github.com/leanprover-community/mathlib/pull/6456/commits/d15691c5f103c0a144239eba35a34662a25e6e36
first the `apply rescale_injective` fails and then  when trying `simpa using mul_right_cancel' hexp h,`, I get 
```
failed to synthesize type class instance for
A : Type u_1,
_inst_1 : comm_ring A,
_inst_2 : no_zero_divisors A,
_inst_3 : nontrivial A,
_inst_4 : algebra ℚ A,
h :
  ⇑eval_neg_hom (mk (λ (n : ℕ), ⇑(algebra_map ℚ A) (bernoulli n / ↑n!))) * (⇑eval_neg_hom (exp A) - 1) *
      -exp A =
    -X * -exp A,
hexp : -exp A ≠ 0
⊢ cancel_monoid_with_zero (power_series A)
state:
A : Type u_1,
_inst_1 : comm_ring A,
_inst_2 : no_zero_divisors A,
_inst_3 : nontrivial A,
_inst_4 : algebra ℚ A,
h :
  ⇑eval_neg_hom (mk (λ (n : ℕ), ⇑(algebra_map ℚ A) (bernoulli n / ↑n!))) * (⇑eval_neg_hom (exp A) - 1) *
      -exp A =
    -X * -exp A,
hexp : -exp A ≠ 0
⊢ ⇑eval_neg_hom (mk (λ (n : ℕ), ⇑(algebra_map ℚ A) (bernoulli n / ↑n!)) * (exp A - 1)) = ⇑eval_neg_hom X
```

Reply:
Sorry for the dead ends - it sounds like what you had before was the best bet after all!

Reply:
no problems, I happy for the comments!
reverting..